src/grdfilter.c

```diff
@@ -584,72 +584,79 @@ GMT_LOCAL struct GMT_GRID *init_area_weights (struct GMT_CTRL *GMT, struct GMT_G
-	GMT_Message (API, GMT_TIME_NONE, "\t        Append +c to center the bins on multiples of <bin> [Default has bin edges as multiples]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t        Append +l to return the lowest mode if multiple modes are found [return average].\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t        Append +u to return the uppermost mode if multiple modes are found [return average].\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     l: Lower : return minimum of all points.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     L: Lower+ : return minimum of all positive points.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     m: Median : return the median (50%% quantile) value of all points.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t        Append +q<quantile> to select another quantile (in 0-1 range) [0.5].\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     p: Maximum likelihood probability estimator : return mode of all points.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t        We approximate the mode using the Least Median of Squares estimator.\n");
+	GMT_Message (API, GMT_TIME_NONE, "  REQUIRED ARGUMENTS:\n");
+	GMT_Usage (API, 1, "\n<ingrid> is the input grid file to be filtered.");
+	GMT_Usage (API, 1, "\n-D<flag>]");
+	GMT_Usage (API, -2, "Distance flag determines how grid (x,y) maps into distance units of filter width as follows, first for Cartesian data:");
+	GMT_Usage (API, 3, "p: grid x,y with <filter_width> in pixels (must be an odd number), Cartesian distances.");
+	GMT_Usage (API, 3, "0: grid x,y same units as <filter_width>, Cartesian distances.");
+	GMT_Usage (API, -2, "Geographic (lon,lat) Data:");
+	GMT_Usage (API, 3, "1: grid x,y in degrees, <filter_width> in km, Cartesian distances.");
+	GMT_Usage (API, 3, "2: grid x,y in degrees, <filter_width> in km, x scaled by cos(middle y), Cartesian distances.");
+	GMT_Usage (API, -2, "The options above are faster; they allow weight matrix to be computed only once. "
+		"Next three options are slower; weights must be recomputed for each scan line.:");
+	GMT_Usage (API, 3, "3: grid x,y in degrees, <filter_width> in km, x scale varies as cos(y), Cartesian distances.");
+	GMT_Usage (API, 3, "4: grid x,y in degrees, <filter_width> in km, spherical distances.");
+	GMT_Usage (API, 3, "5: grid x,y in Mercator units (-Jm1), <filter_width> in km, spherical distances.");
+	GMT_Usage (API, 1, "\n-F<type><filter_width>[/<width2>][<modifiers>]");
+	GMT_Usage (API, -2, "Set the low-pass filter type and full diameter (6 sigma) filter-width. "
+		"Choose between convolution-type filters which differ in how weights are assigned "
+		"and geospatial filters that seek to return a representative value. "
+		"Append +h to select high-pass filtering [Default is low-pass filtering]. "
+		"Filters are isotropic.  For rectangular filtering append /<width2> (requires -Dp|0). "
+		"Convolution filters:");
```

Comment:
```suggestion
	GMT_Usage (API, -2, "Convolution filters:");
```
New line for nicer layout (2/2)

Reply:
DOne